src/librustc_data_structures/stable_map.rs

```diff
@@ -0,0 +1,80 @@
+pub use rustc_hash::FxHashMap;
+use std::borrow::Borrow;
+use std::collections::hash_map::{Entry, IterMut};
+use std::hash::Hash;
+
+/// A deterministic wrapper around FxHashMap that does not provide iteration support.
+///
+/// It supports insert, remove, get and get_mut functions from FxHashMap.
+/// It also allows to convert hashmap to a sorted vector with the function `into_sorted_vector()`.
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct StableMap<K, V>
+where
+    K: Eq + Hash,
+{
+    base: FxHashMap<K, V>,
+}
+
+impl<K: Hash + Eq + Ord, V: Ord> StableMap<K, V> {
+    pub fn new() -> StableMap<K, V> {
+        StableMap { base: FxHashMap::default() }
+    }
+
+    pub fn into_sorted_vector(self) -> Vec<(K, V)>
+    where
+        V: PartialOrd,
+    {
+        let mut v = self.base.into_iter().collect::<Vec<_>>();
+        v.sort_unstable();
```

Comment:
```suggestion
        v.sort_unstable_by_key(|(k, _)| k);
```

That way you dont need `V: PartialOrd`.

Reply:
Thank you very much! I do not much understand the `|(k, _)|` part as of now, could you please explain that? (Just for my learning)

Reply:
Also, I get an error here that trait bound `Ord` is not satisfied after removing `K: Ord`

Reply:
Then the where clause above should be `K: Ord` instead of `K: PartialOrd`.

Reply:
> I do not much understand the |(k, _)| part as of now, could you please explain that?

It is kind of a shorthand for:

```rust
|arg_1| match arg_1 {
    (k, _) => /*...*/
}
```